From: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Date: Mon, 9 Dec 2024 11:04:08 +0000
Subject: [PATCH] mm: fix typos in !memfd inline stub

I typo'd the declaration of memfd_check_seals_mmap() in the case where
CONFIG_MEMFD_CREATE is not defined, resulting in build failures.

Fix this, and correct the misspelling of vm_flags which should be
vm_flags_ptr at the same time.

Link: https://lkml.kernel.org/r/7dee6c5d-480b-4c24-b98e-6fa47dbd8a23@lucifer.local
Signed-off-by: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 include/linux/memfd.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/memfd.h b/include/linux/memfd.h
index d53408b0bd31..246daadbfde8 100644
--- a/include/linux/memfd.h
+++ b/include/linux/memfd.h
@@ -24,7 +24,8 @@ static inline struct folio *memfd_alloc_folio(struct file *memfd, pgoff_t idx)
 {
 	return ERR_PTR(-EINVAL);
 }
-int memfd_check_seals_mmap(struct file *file, unsigned long *vm_flags)
+static inline int memfd_check_seals_mmap(struct file *file,
+					 unsigned long *vm_flags_ptr)
 {
 	return 0;
 }
-- 
2.39.5

